src/Form/EndUserInputNameProvider.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form;
+
+/**
+ * Helpdesk form
+ */
+final class EndUserInputNameProvider
+{
+    public const END_USER_INPUT_NAME = 'answers_%d';
+    public const END_USER_INPUT_NAME_REGEX = '/^_?answers_(\d+)$/';
+
+    /**
+     * Get the end user input name for a given question
+     *
+     * @param Question $question
+     * @return string
+     */
+    public static function getEndUserInputName(Question $question): string
+    {
+        return sprintf(static::END_USER_INPUT_NAME, $question->getID());
+    }
+
+    /**
+     * Get the answers submitted by the end user
+     * The answers are indexed by question ID
+     *
+     * @return array
+     */
+    public static function getAnswers(): array
+    {
+        $filteredAnswers = self::filterAnswers($_POST);
+        $reindexedAnswers = self::reindexAnswers($filteredAnswers);
+
+        return $reindexedAnswers;
+    }
```

Comment:
When dealing with data from $_POST, it is usually better to use an `array $input` parameter in your function.
For exemple, this is cleaner for tests (no need to generate $_POST values).